src/actions/request.rs

```diff
@@ -215,9 +215,33 @@ impl Runnable for Request {
         });
 
         if let Some(&SetCookie(ref cookies)) = response.headers.get::<SetCookie>() {
-          if let Some(cookie) = cookies.iter().next() {
-            let value = String::from(cookie.split(";").next().unwrap());
-            context.insert("cookie".to_string(), Yaml::String(value));
+          let context_cookies = context.entry("cookies".to_string()).or_insert_with(|| Yaml::Array(Vec::new()));
+          if let Yaml::Array(array) = context_cookies {
+            let mut existing = array
+              .iter()
+              .enumerate()
+              .map(|(index, cookie)| {
+                if let Yaml::String(cookie) = cookie {
+                  let key = cookie.split('=').next().unwrap().trim().to_string();
+                  (key, index)
+                } else {
+                  panic!("Unexpected cookie element {:?}", cookie);
+                }
+              })
+              .collect::<HashMap<String, usize>>();
+            for cookie in cookies.iter() {
+              for cook in cookie.split(';') {
+                let key = cook.split('=').next().unwrap().trim();
+                if let Some(index) = existing.get(key) {
+                  array[*index] = Yaml::String(cook.to_string());
```

Comment:
I'm not sure if an array is the best structure for this. With a hash of cookie_name=>cookie_value we can avoid to keep indexes  in these iterations.

Reply:
Maybe, I can switch to `Yaml::Hash` which is backed by a `LinkedHashMap` apparently, this is good for determinism. Do you want that?

Reply:
I think the code is going to be easier because enumerations and indexes disappear.

Reply:
Alright, I'm going to do that.